File: Algorithms/Sorting_Algorithms/Java/QuickSort.java

public class QuickSort {

	private static void swap(int[] arr, int i, int j) {
		int tmp = arr[i];
		arr[i] = arr[j];
		arr[j] = tmp;
	}
	
	private static int partition(int[] arr, int lowerBound, int upperBound) {
		int pivot = arr[upperBound];
		int lb_idx = lowerBound;
		
		for (int j = lb_idx; j < upperBound; j++) 
			if(arr[j] < pivot)
				swap(arr, lb_idx++, j);
		
		swap(arr, lb_idx, upperBound);
		
		return lb_idx;
	}

	private static void quickSort(int[] arr, int lowerBound, int upperBound) {
		
		if(lowerBound < upperBound) {
			int pivot_index = partition(arr, lowerBound, upperBound);
			
			quickSort(arr, lowerBound, pivot_index - 1);
			quickSort(arr, pivot_index + 1, upperBound);
		}
		
	}

	public static void main(String[] args) {
		int array[] = {30, 10, 40, 20, 80, 60, 50, 70};
		int lowerBound = 0;
		int upperBound = array.length - 1;
		quickSort(array, lowerBound, upperBound);
		String sortedArray = java.util.Arrays.toString(array);
		System.out.println(sortedArray);
	}
	
}

File: Algorithms/Sorting_Algorithms/Python/quick_sort.py

def partition(arr, lower_bound, upper_bound):

    pivot = arr[upper_bound]
    lb_idx = lower_bound
    for j in range(lb_idx, upper_bound):
        if arr[j] < pivot:
            arr[lb_idx], arr[j] = arr[j], arr[lb_idx]
            lb_idx += 1
        
    arr[lb_idx], arr[upper_bound] = arr[upper_bound], arr[lb_idx]
    return lb_idx


def quick_sort(arr, lower_bound, upper_bound):
    if lower_bound < upper_bound:

        pivot_index = partition(arr, lower_bound, upper_bound)

        quick_sort(arr, lower_bound, pivot_index - 1)
        quick_sort(arr, pivot_index + 1, upper_bound)


def main():
    vector = [90, 10, 30, 20, 50, 40, 60, 80, 70]
    lower_bound = 0
    upper_bound = len(vector) - 1
    quick_sort(vector, lower_bound, upper_bound)
    print(vector)


if __name__ == "__main__":
    main()